### PR TITLE
script stage setup and documentation

### PR DESCRIPTION
**Pull request by tomaslin**

### CODE REVIEW

**Comment by tomaslin** [comment]
@cfieber please review

File: script_stage/scriptJobConfig.xml [hunk 1]
```diff
@@ -0,0 +1,151 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (cfieber) [inline script_stage/scriptJobConfig.xml]: I'd rather see this as a job-dsl

There's a few things in here that don't really make sense (environment specific)

File: script_stage/scriptJobConfig.xml [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      <permission>hudson.scm.SCM.Tag:api_team</permission>
+      <permission>hudson.model.Item.Configure:api_team</permission>
+      <permission>hudson.model.Run.Update:api_team</permission>
+      <permission>hudson.model.Item.Discover:anonymous</permission>
+    </hudson.security.AuthorizationMatrixProperty>
+    <hudson.plugins.buildblocker.BuildBlockerProperty plugin="build-blocker-plugin@1.7.1">
+      <useBuildBlocker>true</useBuildBlocker>
+      <blockLevel>UNDEFINED</blockLevel>
+      <scanQueueFor>DISABLED</scanQueueFor>
+      <blockingJobs>STASH_MAINTENANCE_DOWNTIME</blockingJobs>
```
Comment (cfieber) [inline script_stage/scriptJobConfig.xml]: remove

File: script_stage/scriptJobConfig.xml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+          <description>Environment Spinnaker is running against</description>
+          <defaultValue></defaultValue>
+        </hudson.model.StringParameterDefinition>
+        <hudson.model.StringParameterDefinition>
+          <name>CLUSTER_PARAM</name>
+          <description>The cluster Spinnaker is deploying to</description>
+          <defaultValue></defaultValue>
+        </hudson.model.StringParameterDefinition>
+        <hudson.model.StringParameterDefinition>
+          <name>CMC</name>
```
Comment (cfieber) [inline script_stage/scriptJobConfig.xml]: feels like we don't populate this through any magic

File: script_stage/scriptJobConfig.xml [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+          <description>The CMC this deployment is associated with</description>
+          <defaultValue></defaultValue>
+        </hudson.model.StringParameterDefinition>
+        <hudson.model.StringParameterDefinition>
+          <name>CONTEXT</name>
```
Comment (cfieber) [inline script_stage/scriptJobConfig.xml]: or this?

File: script_stage/scriptJobConfig.xml [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  <concurrentBuild>true</concurrentBuild>
+  <builders>
+    <hudson.plugins.descriptionsetter.DescriptionSetterBuilder plugin="description-setter@1.10">
+      <regexp></regexp>
+      <description>TASK=${TASK_ID} REGION=${REGION_PARAM} ENV=${ENV_PARAM} CLUSTER=${CLUSTER_PARAM} IMAGE=${IMAGE_ID} CMC=${CMC} ${SCRIPT_PATH}/${COMMAND} ${CONTEXT}</description>
+    </hudson.plugins.descriptionsetter.DescriptionSetterBuilder>
+    <hudson.tasks.Shell>
+      <command>export PATH=$PATH:/apps/jenkins/tools/groovy-2.0.4/bin
+export JAVA_OPTS=&apos;-Dgrape.root=${WORKSPACE}/.groovy/grape&apos;
+export P4USER=rolem
```
Comment (cfieber) [inline script_stage/scriptJobConfig.xml]: perforce baby

File: script_stage/scriptJobConfig.xml [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+  <jdk>Oracle JDK8</jdk>
+  <triggers/>
+  <concurrentBuild>true</concurrentBuild>
+  <builders>
+    <hudson.plugins.descriptionsetter.DescriptionSetterBuilder plugin="description-setter@1.10">
+      <regexp></regexp>
+      <description>TASK=${TASK_ID} REGION=${REGION_PARAM} ENV=${ENV_PARAM} CLUSTER=${CLUSTER_PARAM} IMAGE=${IMAGE_ID} CMC=${CMC} ${SCRIPT_PATH}/${COMMAND} ${CONTEXT}</description>
+    </hudson.plugins.descriptionsetter.DescriptionSetterBuilder>
+    <hudson.tasks.Shell>
+      <command>export PATH=$PATH:/apps/jenkins/tools/groovy-2.0.4/bin
```
Comment (cfieber) [inline script_stage/scriptJobConfig.xml]: groovy path is not generic

File: script_stage/scriptJobConfig.xml [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+      <onlyIfSuccessful>false</onlyIfSuccessful>
+      <fingerprint>false</fingerprint>
+      <defaultExcludes>true</defaultExcludes>
+    </hudson.tasks.ArtifactArchiver>
+  </publishers>
+  <buildWrappers>
+    <EnvInjectBuildWrapper plugin="envinject@1.91.4">
+      <info>
+        <propertiesContent>PYTHONUNBUFFERED=1
+P4USER=rolem</propertiesContent>
```
Comment (cfieber) [inline script_stage/scriptJobConfig.xml]: more p4

**Comment by anotherchrisberry** [comment]
Is this PR still active?

**Comment by tomaslin** [comment]
need to find time to address the issues Cam raised

**Comment by tomaslin** [comment]
closing this until we have a better story for job import

**Pull request closed without merging**